javascript/dwa-starter-vue/src/lib/drls.ts

```diff
@@ -0,0 +1,42 @@
+export const drlFetchRecord = async (did: string, recordId: string) => {
```

Comment:
take a look at the updated one here:
- https://github.com/TBD54566975/tbd-examples/pull/292

also you will need to make sure the activatePolyfills are being called properly from the new `@web5/package` (also can be seen in the above pr).

if you need help doing the changes here please let us know, I will be glad to take it over and help!

Reply:
also @leordev, i tested the changes on my end, and it resulted in the same error as before

Reply:
Reason why you are getting the error is because the web5/browser polyfill seems to not be activated properly: 
![image](https://github.com/user-attachments/assets/5d66b416-0578-4757-b20b-efe877c62941)

this network call should be resolved to the remote dwn... let me keep troubleshooting your branch hold on...

Reply:
Yes same thing for the avatar!

Reply:
I encountered an error when doing the same for the avatar

Reply:
Can you expand on that? Yeah I just pulled your branch and the avatar retrieval does not work... The very same endpoint is failing on the vue side, but in react the same url works...

example:
https://dwn.tbddev.org/beta/did:dht:n98n9d16jhyhugz56hoqzsr4rcjrwpfykdudss98onjecnn7eb9o/read/protocols/aHR0cHM6Ly9zY2hlbWEub3JnL1Byb2ZpbGVTYW1wbGU/avatar

Reply:
@leordev 
<img width="1470" alt="Screenshot 2024-10-18 at 20 00 27" src="https://github.com/user-attachments/assets/af186ef3-512b-4dd8-845c-112ec0fefe86">

Reply:
i read in the docs to fetch by recordId buh would to store the recordId of the created avatar in the local storage then use it to fetch the avatar on component mount

Reply:
> Can you expand on that? Yeah I just pulled your branch and the avatar retrieval does not work... The very same endpoint is failing on the vue side, but in react the same url works...
> 
> example: https://dwn.tbddev.org/beta/did:dht:n98n9d16jhyhugz56hoqzsr4rcjrwpfykdudss98onjecnn7eb9o/read/protocols/aHR0cHM6Ly9zY2hlbWEub3JnL1Byb2ZpbGVTYW1wbGU/avatar

weird, doing more test now

Reply:
Can you double check in your end if you are using the following key steps to upload the avatar image please?

```ts
// assuming you have the image uploaded in a file input field in the variable `file`, of type `File`

// instantiate a blob
const blob = new Blob([file], { type: file.type }) 

// update the avatar in the dwn
const { record, status } = await dwn.records.create({
    data: blob,
    message: {
      published: true,
      recipient: did,
      dataFormat: blob.type,
      protocol: profile.uri,
      protocolPath: "avatar",
    },
  });
```

Reply:
apologies, skipped a few steps in converting the image file to blob, fixed now @leordev @blackgirlbytes